src/codegen/properties/attribute.py

```diff
@@ -0,0 +1,281 @@
+class Attribute():
```

Comment:
What do the parentheses signify? In Python 2, `class Attribute:` and `class Attribute(object):` had subtly different meanings, but AFAIK in Python 3 you can just write `class Attribute:`.

Reply:
As a general notation, If an object is callable it means we will use the round parentheses to make a call to that object. But as you have mentioned, Python 3 removes need for such a notation for python class. Since it doesn't make any difference, I'll remove the parentheses.

Reply:
A class definition is not a function call. The parentheses specify a list of base classes. I guess this list is allowed to be empty, at least in Python 3, but I've never seen code that specified an empty list before.